src/plugins/data/public/search/aggs/buckets/ip_range_fn.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { Assign } from '@kbn/utility-types';
+import { ExpressionFunctionDefinition } from '../../../../../expressions/public';
+import { AggExpressionType, AggExpressionFunctionArgs, BUCKET_TYPES } from '../';
+import { getParsedValue } from '../utils/get_parsed_value';
+
+const fnName = 'aggIpRange';
+
+type Input = any;
+type AggArgs = AggExpressionFunctionArgs<typeof BUCKET_TYPES.IP_RANGE>;
+
+type Arguments = Assign<AggArgs, { ranges?: string }>;
+
+type Output = AggExpressionType;
+type FunctionDefinition = ExpressionFunctionDefinition<typeof fnName, Input, Arguments, Output>;
+
+export const aggIpRange = (): FunctionDefinition => ({
+  name: fnName,
+  help: i18n.translate('data.search.aggs.function.buckets.ipRange.help', {
+    defaultMessage: 'Generates a serialized agg config for a Ip Range agg',
+  }),
+  type: 'agg_type',
+  args: {
+    id: {
+      types: ['string'],
+      help: i18n.translate('data.search.aggs.buckets.ipRange.id.help', {
+        defaultMessage: 'ID for this aggregation',
+      }),
+    },
+    enabled: {
+      types: ['boolean'],
+      default: true,
+      help: i18n.translate('data.search.aggs.buckets.ipRange.enabled.help', {
+        defaultMessage: 'Specifies whether this aggregation should be enabled',
+      }),
+    },
+    schema: {
+      types: ['string'],
+      help: i18n.translate('data.search.aggs.buckets.ipRange.schema.help', {
+        defaultMessage: 'Schema to use for this aggregation',
+      }),
+    },
+    field: {
+      types: ['string'],
+      required: true,
+      help: i18n.translate('data.search.aggs.buckets.ipRange.field.help', {
+        defaultMessage: 'Field to use for this aggregation',
+      }),
+    },
+    ipRangeType: {
+      types: ['string'],
+      required: true,
+      help: i18n.translate('data.search.aggs.buckets.ipRange.ipRangeType.help', {
+        defaultMessage:
+          'Ip range type to use for this aggregation. Takes one of the following values: mask, fromTo.',
+      }),
+    },
```

Comment:
We should add `options` here (which I'm realizing I forgot to do for the `order` param in `terms`):

```suggestion
    ipRangeType: {
      types: ['string'],
      required: true,
      options: ['mask', 'fromTo'],
      help: i18n.translate('data.search.aggs.buckets.ipRange.ipRangeType.help', {
        defaultMessage:
          'IP range type to use for this aggregation. Takes one of the following values: mask, fromTo.',
      }),
    },
```